src/main/java/org/dcsa/ebl/Util.java

```diff
@@ -18,6 +20,20 @@
      */
     public static final int SQL_LIST_BUFFER_SIZE = 70;
 
+    public static <T, I> void checkForDuplicates(Iterable<T> objects, Function<T, I> idMapper, String name) {
+        Set<I> usedIds = new HashSet<>();
+        for (T t : objects) {
+            I id = idMapper.apply(t);
+            if (id == null) {
+                continue;
+            }
+            if (!usedIds.add(id)) {
+                throw new UpdateException("Duplicate ID " + id + " (" + name + "):  The ID must occur at most once.");

```

Comment:
Do you want this to be a general function or a function that is only to check for duplicate IDs?
If it is a general function then the error message should not contain the wording "ID"

Reply:
It is only used for IDs (or de facto IDs like equipment reference).